src/middy.js

```diff
@@ -72,7 +72,7 @@ const runErrorMiddlewares = (middlewares, instance, done) => {
   const runNext = (err) => {
     try {
       if (!err) {
-        return done()
+        instance.__handledError = true
```

Comment:
I am thinking now that we should probably reset that flag at some point. Let me review this before moving the PR forward